### PR TITLE
Use tiled-api npm package

### PR DESCRIPTION
**Pull request by nerochiaro**
Attempt to implement #37 by adding a `package.json` and using JSDoc to add missing type information, as suggested.

Note that the branch behind this PR builds on top of the one from #48, as otherwise vscode would have trouble understanding where the functions from utils come from, since they are not explicitly imported.
Also, with the renaming of the files to `.mjs` the merge would be messier if I had based this branch on master.

Also note that the values in `package.json` need to be checked by someone with more experience about the history of the project, especially author and license. The original project had no `LICENSE` file or other license information, so I put GPLv2 as it's what is used for Tiled itself, but I'm probably wrong.

The value in the package version is the version number of the most recent release on github of the exporter.

Finally, a few notes: after merging this branch if you add `// @ts-check` to the top of the files, you will enable static type checking, which highlights quite a few interesting potential issues.
I don't know if it's best to submit separate bug reports for these, or simply attempt to fix the obvious ones in another PR.

### CODE REVIEW

**Comment by eleniums** [comment]
This comparison shows the changes in this PR compared to #48, which this one is built off of:
https://github.com/nerochiaro/tiled-to-godot-export/compare/use_modules...use_types

File: package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "tiled-to-godot-export",
+  "version": "1.8.0",
+  "description": "Tiled plugins for exporting Tilemaps and Tilesets in Godot 3.x format",
+  "main": "export_to_godot_tilemap.js",
```
Comment (eleniums) [inline package.json:5]: I think this should be updated to `export_to_godot_tilemap.mjs`.

File: .gitignore [hunk 2]
```diff
@@ -49,3 +49,10 @@ Desktop.ini
 *.log*
 
 /.idea/
+
+######################
+# NPM
+######################
+
+/node_modules/
+package-lock.json
```
Comment (eleniums) [inline .gitignore:58]: The `package-lock.json` file should actually be committed and not ignored.

**Review by eleniums — COMMENTED** [review]
Thanks for looking into this! It seems to work as I can see Tiled API code completion after I run `npm install`.

Can you please update the file extension in the readme, since files now have the extension `.mjs` instead of `.js`?

Perhaps @bjorn or @MikeMnD can comment on whether GPLv2 is the appropriate license to use here.

**Comment by MikeMnD** [comment]
About the license my intention was to be free, so license should be the same as Tiled or Godot.
Since this plugin is more of a part ot Tiled I think GPLv2 is the right one. :)

**Review by eleniums — APPROVED** [review]
Changes look good. Thanks!

**Comment by bjorn** [comment]
That looks great, thanks @nerochiaro!

**Pull request merged**